core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -315,6 +315,34 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * Gets or computes an RDD partition. Used by RDD.iterator() when an RDD is cached.
```

Comment:
Can you add that this is only called on the executors? (otherwise people might wonder why we don't just use `sc.env` instead of `SparkEnv.get`)